### PR TITLE
How to collect update and organize python logs

### LINKED ISSUE #4635: [Python] How to collect, update, and organize Python logs
#### Proposed title of the article
[Python]How to collect, update, and organize Python logs
#### Proposed article introduction
 The built-in logging module in Python is intended to provide vital visibility into your applications with little configuration. This article describes how to configure Python's logging module to log all the data needed, route it to desired destinations, and consolidate logs to gain deeper insights into Python applications.
 An external configuration file can also be used to configure the Python logging subsystem. The Python standard library contains the specifications for the logging configuration format.
 The logging library follows a modular design and is divided into four categories: loggers, handlers, filters, and formats.
#### Key takeaways 
1. Customize your logs' priority level and destination.
2. Create a customized setup with various loggers and destinations.
3. Include tracebacks and exception handling in your logs.
4. To facilitate troubleshooting, format your logs in JSON and consolidate them.
#### Article quality
The article thoroughly explains how Python logs work, providing students with a solid foundation in logging. It will enhance their skills since every detail is put in place.
#### References
N/A

### PR DESCRIPTION
**Pull request by kymomolly**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4635

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by kymomolly** [comment]
@paulodhiambo, have polished up my article you may have a look at it when free..

**Comment by LinusMuema** [comment]
Hi @kymomolly 

Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[How to collect update and organize python logs #5146 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7608325/How.to.collect.update.and.organize.python.logs.5146.Copyleaks.Report.pdf)

**Comment by kymomolly** [comment]
@paulodhiambo, it seems some of the indicated  parts  that are highlighted are functions and cannot be changed.

**Comment by kymomolly** [comment]
@paulodhiambo, I've the required updates and I hope the article is ready to move to the next step.

**Comment by kymomolly** [comment]
@paulodhiambo, how is the progress

**Comment by paulodhiambo** [comment]
@kymomolly please format your code snippets

**Comment by kymomolly** [comment]
@paulodhiambo , I've done the code formatting you have a check on

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+logging.config.indexConfig('/way/to/logging.ini', disable_existing_loggers=Faulty) 
+logger = logging.getLogger(__name__) 
+def word_count(myid): 
+endeavor: 
+#remember the number of words for a record, file, and log the result 
+[...] 
+be that as it may, OSError as e: 
+logger.error(e) 
+logger.error(e, exc_info=True) 
+[...]
```
Comment (paulodhiambo) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:187]: please format this section

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+'filename': 'dataflair-debug.log', 
+}, 
+'console': { 
+'class': 'logging.StreamHandler', 
+}, 
+}, 
+#Loggers appended
+'lumberjacks': { 
+'django': { 
+'overseers': ['file', 'console'],
```
Comment (paulodhiambo) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:158]: Format this section

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+#use the try except block to configure multiple logs
+try: 
+word_count = lessermodule.word_count(myid) 
+with open('frametallyarchive.csv', 'a') as record: 
+line = str(myid) + ',' + str(word_count) 
+index.write(row + '\n') 
+except: 
+logger.warning("unable to generate report%s to impartial", myid) 
+finally: 
+logger.debug("the task is ramshackle the report %s", myid)
```
Comment (paulodhiambo) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md:62]: Format this section

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@kymomolly Please format your code sections, most of the readers copy and paste code directly from the article. With current formatting, if someone copies and pastes your code snippet, then it will not work because of the formatting.

**Comment by kymomolly** [comment]
@paulodhiambo, I have done the requested changes. You can have a look.

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md (hunk 3 above)
Comment (kymomolly) [reply content/articles/how-to-collect-update-and-organize-python-logs/index.md:54]: Done formatting

**Comment by kymomolly** [comment]
Hello @paulodhiambo , I've done the right formatting of the code snippets ,have a look.

**Comment by hectorkambow** [comment]
https://github.com/section-engineering-education/engineering-education/pull/4823

**Comment by hectorkambow** [comment]
@paulodhiambo  

This is a duplicate PR - please keep that in mind when performing a review.

File: content/articles/how-to-collect-update-and-organize-python-logs/index.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    except OSError as e:
+        logger.error(e)
+        logger.error(e, exc_info=True)
+```
+- Recording unhandled exceptional cases: 
+It isn't easy to go before and resolve each achievable exceptional case. But, fundamentally, guarantee your logs can get every exclusion, and you can deal with them later. 
+An unhandled exclusion happens when the application code doesn't true to form handle extraordinary cases that are outside the `try...except` block. For instance, when you endeavor to open a record on a plate, it's anything but a surprising issue for the archive not to exist. The .NET Framework will then throw a 'FileNotFoundException'.  More to that you can use python's standard [traceback lirary](https://docs.python.org/3/library/traceback.html) to coordinate traceback and connect it in the log message in an event as under:
+```python 
+
+```
```
Comment (paulodhiambo) [inline content/articles/how-to-collect-update-and-organize-python-logs/index.md]: @kymomolly is there code snippet missing here?

Comment (kymomolly) [reply content/articles/how-to-collect-update-and-organize-python-logs/index.md]: > @kymomolly is there code snippet missing here?

@paulodhiambo , I've added the code

**Comment by paulodhiambo** [comment]
@kymomolly go through the article and rectify the grammar errors you caused when editing the article. For any help, ping me on slack. I'll be closing this article for now because it's a bit far from a polished article that can be published.

**Comment by molly-kymo** [comment]
Hello @paulodhiambo , you can reopen the article. I'm with grammar fix

**Comment by kymomolly** [comment]
Hello @paulodhiambo , you can reopen the article. I'm done with grammar fix

**Comment by kymomolly** [comment]
Hello @paulodhiambo , you can reopen the article. I'm done with grammar fix

**Comment by paulodhiambo** [comment]
@kymomolly please move the article to the correct directory. Go through your article, I noticed you use words without checking their meaning. Please don't just through vocabulary to your article without checking the actual meaning. I will close this PR for now. Once you are done with the revisions, please open a new PR.

**Pull request closed without merging**